presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/rest/RestSqlFunctionExecutor.java

```diff
@@ -0,0 +1,164 @@
+        checkState(this.blockEncodingSerde == null, "blockEncodingSerde already set");
+        requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
+        this.blockEncodingSerde = blockEncodingSerde;
+        this.pageSerde = new PagesSerde(blockEncodingSerde, Optional.empty(), Optional.empty(), Optional.empty());
+    }
+
+    @Override
+    public CompletableFuture<SqlFunctionResult> executeFunction(
+            String source,
+            RemoteScalarFunctionImplementation functionImplementation,
+            Page input,
+            List<Integer> channels,
+            List<Type> argumentTypes,
+            Type returnType)
+    {
+        SqlFunctionHandle functionHandle = functionImplementation.getFunctionHandle();
+        SqlFunctionId functionId = functionHandle.getFunctionId();
+        DynamicSliceOutput sliceOutput = new DynamicSliceOutput((int) input.getRetainedSizeInBytes());
+        writeSerializedPage(sliceOutput, pageSerde.serialize(input));
+        RequestBody body = RequestBody.create(MediaType.parse("text/plain; charset=utf-8"), sliceOutput.slice().byteArray());
+        Request request = new Request.Builder()
+                .url(getNativeWorkerUri(nodeManager, functionId, returnType).toString())
+                .post(body)
+                .build();
+        CallbackFuture future = new CallbackFuture();
+        okHttpClient.newCall(request).enqueue(future);
+        return future;
+    }
+
+    private URI getNativeWorkerUri(NodeManager nodeManager, SqlFunctionId functionId, Type returnType)
+    {
+        List<String> functionArgumentTypes = functionId.getArgumentTypes().stream().map(TypeSignature::toString).collect(toImmutableList());
+        if (restURIManager.getRestUriManager() == null) {
+            throw new PrestoException(NOT_FOUND, "failed to find native node !");
+        }
+        HttpUriBuilder uri = uriBuilderFrom(URI.create("http://" + restURIManager.getRestUriManager()))
+                .appendPath("/v1/function/" + functionId.getFunctionName().getObjectName())
+                .addParameter("returnType", returnType.toString())
+                .addParameter("numArgs", "" + functionArgumentTypes.size());
+        for (int i = 1; i <= functionArgumentTypes.size(); i++) {
+            uri.addParameter("argType" + i, functionArgumentTypes.get(i - 1));
+        }
+        return uri.build();
+    }
+
+    private class CallbackFuture
+            extends CompletableFuture<SqlFunctionResult>
+            implements Callback
+    {
+        @Override
+        public void onFailure(Call call, IOException e)
+        {
+            super.completeExceptionally(e);
+        }
+        @Override
+        public void onResponse(Call call, Response response)
+        {
+            try {
+                if (response.code() != 200) {
+                    super.completeExceptionally(new IllegalStateException("Failed to get response for rest function call. Response code: " + response.code()));
```

Comment:
Create a new PrestoException with a new error code indicating the function server had a problem